test/e2e/upgradetest/framework/framework.go

```diff
@@ -98,12 +108,30 @@ func (f *Framework) CreateOperator() error {
 								ValueFrom: &v1.EnvVarSource{FieldRef: &v1.ObjectFieldSelector{FieldPath: "metadata.name"}},
 							},
 						},
+						ReadinessProbe: &v1.Probe{
+							Handler: v1.Handler{
+								HTTPGet: &v1.HTTPGetAction{
+									Path: probe.HTTPReadyzEndpoint,
+									Port: intstr.IntOrString{Type: intstr.Int, IntVal: 8080},
+								},
+							},
+							InitialDelaySeconds: 3,
+							PeriodSeconds:       3,
+						},
 					}},
 				},
 			},
 		},
 	}
 	_, err := f.KubeCli.AppsV1beta1().Deployments(f.KubeNS).Create(d)
+	if err != nil {
+		return fmt.Errorf("failed to create deployment: %v", err)
+	}
+
+	lo := metav1.ListOptions{
+		LabelSelector: labels.SelectorFromSet(map[string]string{"name": "etcd-operator"}).String(),
```

Comment:
We should also refactor the code to unify all this label selectors in upgrade test